lib/record_store/cli.rb

```diff
@@ -103,9 +103,11 @@ def apply
         exit
       end
 
-      zones.each do |zone|
-        abort("Attempted to apply invalid zone: #{zone.name}: #{zone.errors.full_messages.join(',')}") unless zone.valid?
+      if (invalid_zone = zones.find(&:invalid?))
+        abort("Attempted to apply invalid zone: #{invalid_zone.name}: #{invalid_zone.errors.full_messages.join(',')}")
+      end
```

Comment:
One thing that would be nice to avoid back-and-forth here would be to go through that loop, list all invalid zones, and then abort with one message listing all the invalid zones right away?
That way, if I make two mistakes, I have the two errors right away, instead of needing to fix one to see the other.

Reply:
Ok - as per @cttttt's [comment](https://github.com/Shopify/record_store/pull/155#discussion_r401693619) I'll revert out all changes to cli.rb & make a follow-up PR with your suggestion to list all invalid zones =)